src/core/agent/services/connectionService.ts

```diff
@@ -252,10 +270,81 @@ class ConnectionService extends AgentService {
 
   @OnlineOnly
   async deleteConnectionById(id: string): Promise<void> {
-    await this.props.signifyClient.contacts().delete(id);
+    await this.props.signifyClient
+      .contacts()
+      .delete(id)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (!/404/gi.test(status)) {
+          throw error;
+        }
+      });
     await this.connectionStorage.deleteById(id);
   }
 
+  async markConnectionPendingDelete(id: string) {
+    const connectionProps = await this.connectionStorage.findById(id);
+    if (!connectionProps) return;
+    connectionProps.pendingDeletion = true;
+    await this.connectionStorage.update(connectionProps);
+
+    this.props.eventEmitter.emit<ConnectionRemovedEvent>({
+      type: EventTypes.ConnectionRemoved,
+      payload: {
+        connectionId: id,
+      },
+    });
+  }
+
+  async getConnectionsPendingDeletion() {
+    const connections = await this.connectionStorage.findAllByQuery({
+      pendingDeletion: true,
+    });
+
+    return connections.map((connection) => connection.id);
+  }
+
+  async getConnectionsPending() {
+    const connections = await this.connectionStorage.findAllByQuery({
+      pending: true,
+      groupId: undefined,
+    });
+
+    return connections.map((connection) => connection.id);
+  }
+
+  @OnlineOnly
+  async reconnectPendingConnectionById(id: string) {
+    const connection = await this.props.signifyClient
+      .contacts()
+      .get(id)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (/404/gi.test(status)) {
+          throw new Error(`${Agent.MISSING_DATA_ON_KERIA}: ${id}`);
+        } else {
+          throw error;
+        }
+      });
+
+    const operation = await this.resolveOobi(connection.oobi, true);
+    if (operation.done) {
```

Comment:
We need to track this as a proper long running operation. Store it, and when it's done, mark as complete.

Reply:
this may conflict with Martin's PR too. When we are resolving a normal connection now - I don't want to wait 2 seconds. Just immediately track the long running operation in the DB. Please coordinate there

Reply:
I discussed this with Martin, it is almost done. After that, I will point this PR to here.

Reply:
OK, this function needs to cmopletely change. It just resolves the OOBI and stores the long running operation. After that, you need to track the long running operation to complete and move this logic there (e.g. connectionProps.pending = false)`. You should be able to do that already